ethers-contract/ethers-contract-abigen/src/contract/events.rs

```diff
@@ -10,11 +10,12 @@ use syn::Path;
 impl Context {
     /// Expands each event to a struct + its impl Detokenize block
     pub fn events_declaration(&self) -> Result<TokenStream> {
+        let mut event_aliases = self.event_aliases.clone();
         let sorted_events: BTreeMap<_, _> = self.abi.events.clone().into_iter().collect();
         let data_types = sorted_events
             .values()
             .flatten()
-            .map(|event| self.expand_event(event))
+            .map(|event| self.expand_event(event, event_aliases.remove(&event.abi_signature())))
```

Comment:
Since `fn expand_event(&self` has access to `self.event_aliases` we don't need this parameter.
We can instead do in `expand_event`:
`let alias = self.event_aliases.get(&event.abi_signature())`